tests/black_compatibility_test.rs

```diff
@@ -0,0 +1,186 @@
+use std::io::{ErrorKind, Read};
+use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4, TcpListener, TcpStream};
+use std::path::{Path, PathBuf};
+use std::process::Stdio;
+use std::thread::sleep;
+use std::time::Duration;
+use std::{fs, process, str};
+
+use anyhow::{anyhow, Context, Result};
+use assert_cmd::{crate_name, Command};
+use itertools::Itertools;
+use ruff::checks::CheckCategory;
+use strum::IntoEnumIterator;
+use walkdir::WalkDir;
+
+/// Handles blackd process and allows submitting code to it for formatting.
+struct Blackd {
+    address: SocketAddr,
+    server: process::Child,
+    client: ureq::Agent,
+}
+
+impl Blackd {
```

Comment:
Do you have a sense for how much of a difference `blackd` vs. `black` makes?

Reply:
Did a couple runs with blackd vs. black on a Ryzen 5800X. ~44s with blackd, 64s with black.

Reply:
Ok cool. My only hesitation is around robustness. If we see flakiness on CI (e.g., from waiting for `blackd` to startup, or other daemon-related issues), I may ask that we convert back to `black`, since I'd rather CI be reliable than fast (at least, on that order of magnitude).